File: backend/data/generated-driver-examples/reverse-linked-list/driver.py
import sys
import json
from typing import List, Optional, Any

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def parse_linked_list(data: List[int]) -> Optional[ListNode]:
    if not data:
        return None
    dummy = ListNode(0)
    curr = dummy
    for val in data:
        curr.next = ListNode(val)
        curr = curr.next
    return dummy.next

def print_linked_list(head: Optional[ListNode]) -> None:
    res = []
    curr = head
    while curr:
        res.append(curr.val)
        curr = curr.next
    print(json.dumps(res))


def main():
    lines = [line.strip() for line in sys.stdin if line.strip()]
    if not lines:
        return
    
    solution = Solution()
    idx = 0
    while idx < len(lines):
        head = parse_linked_list(json.loads(lines[idx + 0]))
        result = solution.reverseList(head)
        print_linked_list(result)
        idx += 1

if __name__ == "__main__":
    main()

File: backend/data/generated-driver-examples/reverse-linked-list/driver.js
const fs = require('fs');

function ListNode(val, next) {
  this.val = (val === undefined ? 0 : val);
  this.next = (next === undefined ? null : next);
}

function parseLinkedList(arr) {
  if (!arr || arr.length === 0) return null;
  const dummy = new ListNode(0);
  let curr = dummy;
  for (const val of arr) {
    curr.next = new ListNode(val);
    curr = curr.next;
  }
  return dummy.next;
}

function printLinkedList(head) {
  const res = [];
  let curr = head;
  while (curr) {
    res.push(curr.val);
    curr = curr.next;
  }
  console.log(JSON.stringify(res));
}

function main() {
  const input = fs.readFileSync(0, 'utf-8');
  const lines = input.split('\n').map(l => l.trim()).filter(l => l.length > 0);
  if (lines.length === 0) return;

  let idx = 0;
  while (idx < lines.length) {
    var head = parseLinkedList(JSON.parse(lines[idx + 0]));
    var result = reverseList(head);
    printLinkedList(result);
    idx += 1;
  }
}

main();